packages/store/addon/-private/system/snapshot.js

```diff
@@ -5,9 +5,19 @@ import { inspect } from '@ember/debug';
 import EmberError from '@ember/error';
 import { get } from '@ember/object';
 import { assign } from '@ember/polyfills';
-import recordDataFor, { relationshipStateFor } from './record-data-for';
+import recordDataFor from './record-data-for';
 import { CUSTOM_MODEL_CLASS } from '@ember-data/canary-features';
 
+function relationshipsFor(instance) {
+  let recordData = recordDataFor(instance) || instance;
```

Comment:
This fallback seems vaguely odd, why do we need it?

Reply:
we may have received

- a recordData
- a snapshot
- an internalModel
- one of a host of other things

some of this has improved since we originally wrote this util but we should revisit soon

Reply:
Gotcha, seems like we need this as typescript so that set of overloads is clearer.

Reply:
If I've followed the PR correctly, we end up with three identical copies of this function. Is that *desired* here, and if so, do you have a strategy in mind for further refactoring to get that down to a single (typed) implementation in the future?